waylock: init at 0.4.2

jordanisaacs opened the pull request:
###### Description of changes

Initialize [waylock](https://github.com/ifreund/waylock), a small screenlocker for wayland utilizing ext-session-lock-v1. Written by the creator of the river compositor. When using, ensure to install a waylock pam.d file. On NixOS use `security.pam.services.waylock = {}`. Also added myself as a new maintainer!

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Built it in nix packages and ran the binary on my personal system with sway (after including the necessary pam.d file). It worked successfully at locking/unlocking.

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

teutat3s reviewed (commented):
Please use a separate commit to add yourself to the list of maintainers, see: https://nixos.org/manual/nixpkgs/stable/#var-meta-maintainers

teutat3s commented:
Result of `nixpkgs-review pr 201981` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>waylock</li>
  </ul>
</details>

jordanisaacs committed:
maintainers: add jordanisaacs

jordanisaacs committed:
waylock: init at 0.4.2

When using, ensure to install a waylock pam.d file.
On NixOS use `security.pam.services.waylock = {}`.

jordanisaacs commented:
> Please use a separate commit to add yourself to the list of maintainers, see: https://nixos.org/manual/nixpkgs/stable/#var-meta-maintainers

Fixed.
